### PR TITLE
Fix BTN when air by date but show is indexed with SxxExx

### PR DESCRIPTION
**Pull request by duramato**
- [ ] PR is based on the DEVELOP branch
- [ ] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [ ] Read the [contribution guide](https://github.com/pymedusa/Medusa/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: medusa/providers/torrent/json/btn.py [hunk 1]
```diff
@@ -227,6 +227,10 @@ def _episode_search_params(self, ep_obj):
             # BTN uses dots in dates, we just search for the date since that
             # combined with the series identifier should result in just one episode
             search_params['name'] = date_str.replace('-', '.')
+            to_return.append(search_params)
+            search_params = {'category': 'Episode'}
+            search_params['name'] = '{ep}'.format(ep=episode_num(ep_obj.season, ep_obj.episode))
```
Comment (labrys) [inline medusa/providers/torrent/json/btn.py]: You are overwriting `search_params['name']` defined at L229

**Comment by duramato** [comment]
```
2017-02-12 12:32:38 INFO     SEARCHQUEUE-MANUAL-72231 :: [BTN] :: [2bb329d] Performing episode search for Real Time with Bill Maher
2017-02-12 12:32:38 DEBUG    SEARCHQUEUE-MANUAL-72231 :: [BTN] :: [2bb329d] Search mode: Episode
2017-02-12 12:32:38 ERROR    SEARCHQUEUE-MANUAL-72231 :: [BTN] :: [2bb329d] Unknown exception while searching BTN. Error: 'must be string, not datetime.date'
Traceback (most recent call last):
  File "/home/pi/Medusa/medusa/search/core.py", line 527, in search_providers
    down_cur_quality, manual_search, manual_search_type)
  File "/home/pi/Medusa/medusa/providers/generic_provider.py", line 188, in find_search_results
    items_list += self.search(search_string, ep_obj=episode)
  File "/home/pi/Medusa/medusa/providers/torrent/json/btn.py", line 76, in search
    search_params = self._search_params(ep_obj, mode)
  File "/home/pi/Medusa/medusa/providers/torrent/json/btn.py", line 187, in _search_params
    air_date = datetime.strptime(ep_obj.airdate, '%Y-%m-%d')
TypeError: must be string, not datetime.date
2017-02-12 12:32:39 DEBUG    SEARCHQUEUE-MANUAL-72231 :: [BTN] :: [2bb329d] Traceback (most recent call last):
  File "/home/pi/Medusa/medusa/search/core.py", line 527, in search_providers
    down_cur_quality, manual_search, manual_search_type)
  File "/home/pi/Medusa/medusa/providers/generic_provider.py", line 188, in find_search_results
    items_list += self.search(search_string, ep_obj=episode)
  File "/home/pi/Medusa/medusa/providers/torrent/json/btn.py", line 76, in search
    search_params = self._search_params(ep_obj, mode)
  File "/home/pi/Medusa/medusa/providers/torrent/json/btn.py", line 187, in _search_params
    air_date = datetime.strptime(ep_obj.airdate, '%Y-%m-%d')
TypeError: must be string, not datetime.date
```

**Comment by duramato** [comment]
```
2017-02-12 21:59:50 INFO     SEARCHQUEUE-MANUAL-72231 :: [BTN] :: [7920ada] Performing episode search for Real Time with Bill Maher
2017-02-12 21:59:50 DEBUG    SEARCHQUEUE-MANUAL-72231 :: [BTN] :: [7920ada] Search mode: Episode
2017-02-12 21:59:50 DEBUG    SEARCHQUEUE-MANUAL-72231 :: [BTN] :: [7920ada] Search string: [{u'category': u'Episode', u'tvdb': 72231, u'name': '2017.02.10'}, {u'category': u'Episode', u'tvdb': 72231, u'name': u'S15E04'}]
2017-02-12 21:59:50 WARNING  SEARCHQUEUE-MANUAL-72231 :: [BTN] :: [7920ada] Error while accessing provider. Error: No JSON object could be decoded
2017-02-12 21:59:50 DEBUG    SEARCHQUEUE-MANUAL-72231 :: [BTN] :: [7920ada] No data returned from provider
2017-02-12 21:59:50 DEBUG    SEARCHQUEUE-MANUAL-72231 :: [BTN] :: [7920ada] Fallback season pack search initiate
2017-02-12 21:59:50 INFO     SEARCHQUEUE-MANUAL-72231 :: [BTN] :: [7920ada] Performing season pack search for Real Time with Bill Maher
```

Needs to be made in seperate calls i guess, becouse they should have returned results.

**Comment by duramato** [comment]
Can't Approve becouse it's my own PR, but all good.

**Pull request merged**